code/data_structures/src/avl_tree

```diff
@@ -0,0 +1,161 @@
+// Implementation of AVL tree
+// following is code for insertion of data in AVL tree
+#include <bits/stdc++.h> 
+using namespace std;
+// AVL tree node
+struct Node 
+{
+    int data;
+    Node *left;
+    Node *right;
+    int height;
+};
+ // A function which creates new node and assigns given value to node
+struct Node *NewNode(int key)
+{
+    struct Node *temp=(struct Node*)malloc(sizeof(struct Node));
+    temp->data=key;
+    temp->left=temp->right=NULL;
+    return temp;
+}
+// A fuction which calculates height of respective node
+int height(struct Node *node) 
+{
+    if(node==NULL)
+    {return 0;}
+    
+    int lh=height(node->left);
+    int rh=height(node->right);
+    
+    if(lh > rh)
+    {return lh+1;}
+    else
+    {return rh+1;}
+}
+ // A fuction which returns maximum of two numbers
+int max(int a,int b)
```

Comment:
No need to define your own, use `std::max`. This can also create conflict if `using namespace std` is enabled, another reason why you shouldn't do that.